### DISCUSSION TITLE
Very low latence Legrand & Sonoff Dongle E

### DISCUSSION THREAD

**Discussion by Loic691** [Koenkk/zigbee2mqtt#15407] [Devices]
Hi community
Hi @Hedda @kirovilya 
I think I have a new issue with Z2M and Sonoff Dongle E based on EFR32MG21 chip
I have 15 devices wich are working well but I have a new issues with 1  legrand wireless switch 67744.
It responds with a big latence and not all the time.
The button is just near (3m) a router and just  side by side with a roller shutter wireless switch (different model from my issue switch) wich is working well
I do not have this problem 1 month ago. Nothing changes on my installation but just upgrade to Z2M 1.28.4.
I'm not sure this is 1.28.4 because old one have the same issue but not sure what version is 1.28. 0 ou 1.27.2 ?

You can find log trace with 1 push to RIGHT ON button at 00:01:13
The trace show we have very quickly severals zigbee message but not the GenOnOff action which can be arrived 1 minute later, maybe more...
At differement moment, I have simply no zigbee message or exactly they arrived one or two minutes after.

What is your opinion of this issue ?

Info 2022-12-07 00:01:13MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:13+01:00","linkquality":236}'

Info 2022-12-07 00:01:13MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:13+01:00","linkquality":236}'

Debug 2022-12-07 00:01:13Received Zigbee message from 'btn_double_cuisine', type 'raw', cluster 'manuSpecificUbisysDimmerSetup', data '{"data":[21,33,16,1,9,245,44,126,0,0,116,4,0,1],"type":"Buffer"}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:13MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:13+01:00","linkquality":236}'

Debug 2022-12-07 00:01:13Received Zigbee message from 'btn_double_cuisine', type 'raw', cluster 'manuSpecificUbisysDimmerSetup', data '{"data":[21,33,16,2,9,245,44,126,0,0,116,4,0,2],"type":"Buffer"}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:13MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:13+01:00","linkquality":236}'

Debug 2022-12-07 00:01:14Received Zigbee message from 'btn_double_cuisine', type 'raw', cluster 'manuSpecificUbisysDimmerSetup', data '{"data":[21,33,16,2,9,245,44,126,0,0,116,4,0,2],"type":"Buffer"}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:14MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:15+01:00","linkquality":236}'

Debug 2022-12-07 00:01:15Received Zigbee message from 'btn_double_cuisine', type 'raw', cluster 'manuSpecificUbisysDimmerSetup', data '{"data":[21,33,16,1,9,245,44,126,0,0,116,4,0,1],"type":"Buffer"}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:15MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:15+01:00","linkquality":236}'

Debug 2022-12-07 00:01:19Received Zigbee message from 'btn_double_cuisine', type 'raw', cluster 'manuSpecificUbisysDimmerSetup', data '{"data":[21,33,16,2,9,245,44,126,0,0,116,4,0,2],"type":"Buffer"}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:19MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:19+01:00","linkquality":240}'

Debug 2022-12-07 00:01:19Received Zigbee message from 'btn_double_cuisine', type 'raw', cluster 'manuSpecificUbisysDimmerSetup', data '{"data":[21,33,16,1,9,245,44,126,0,0,116,4,0,1],"type":"Buffer"}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:19MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:19+01:00","linkquality":236}'

Debug 2022-12-07 00:01:26Received Zigbee message from 'btn_double_cuisine', type 'raw', cluster 'manuSpecificUbisysDimmerSetup', data '{"data":[21,33,16,2,9,245,44,126,0,0,116,4,0,2],"type":"Buffer"}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:26MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:26+01:00","linkquality":236}'

Debug 2022-12-07 00:01:27Received Zigbee message from 'btn_double_cuisine', type 'raw', cluster 'manuSpecificUbisysDimmerSetup', data '{"data":[21,33,16,1,9,245,44,126,0,0,116,4,0,1],"type":"Buffer"}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:27MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":null,"battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:27+01:00","linkquality":240}'

Debug 2022-12-07 00:01:42Received Zigbee message from 'btn_double_cuisine', type 'commandOn', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:42MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":"on_left","battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:42+01:00","linkquality":236}'
Info 2022-12-07 00:01:42MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":"","battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:42+01:00","linkquality":236}'
Info 2022-12-07 00:01:42MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine/action', payload 'on_left'

Debug 2022-12-07 00:01:44Received Zigbee message from 'btn_double_cuisine', type 'commandOn', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 0
Info 2022-12-07 00:01:44MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":"on_left","battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:44+01:00","linkquality":236}'

Info 2022-12-07 00:01:44MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":"","battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:44+01:00","linkquality":236}'
Info 2022-12-07 00:01:44MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine/action', payload 'on_left'
Debug 2022-12-07 00:01:47Received Zigbee message from 'btn_double_cuisine', type 'commandOn', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 0

Info 2022-12-07 00:01:47MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":"on_left","battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:47+01:00","linkquality":236}'
Info 2022-12-07 00:01:47MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":"","battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:47+01:00","linkquality":236}'
Info 2022-12-07 00:01:47MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine/action', payload 'on_left'
Debug 2022-12-07 00:01:53Received Zigbee message from 'btn_double_cuisine', type 'commandOn', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 0

Info 2022-12-07 00:01:53MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":"on_left","battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:54+01:00","linkquality":236}'
Info 2022-12-07 00:01:53MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine', payload '{"action":"","battery":null,"device":{"applicationVersion":0,"dateCode":" \u000020200902\u0000\u0000\u0000\u0000\u0000","friendlyName":"btn_double_cuisine","hardwareVersion":3,"ieeeAddr":"0x00047400007e2cf5","manufacturerID":4129,"manufacturerName":" Legrand\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","model":"067774","networkAddress":38872,"powerSource":"Battery","softwareBuildID":"0006","stackVersion":66,"type":"EndDevice","zclVersion":2},"last_seen":"2022-12-07T00:01:54+01:00","linkquality":236}'
Info 2022-12-07 00:01:53MQTT publish: topic 'zigbee2mqtt/btn_double_cuisine/action', payload 'on_left'

**Comment by Loic691** [comment]
more simply just with zigbee received message
The physical button was pushed at time of the first line
The GenOnOff was received 40 seconds later....

![image](https://user-images.githubusercontent.com/60837526/206052145-920b0e6d-d411-4efc-adeb-d8903a640fb8.png)

**Comment by jiquem** [comment]
Bind the Legrand to an empty group

**Reply by villmatt** [reply]
hi @jiquem 

Should every light switch be paired with a new empty group?

**Reply by Loic691** [reply]
> hi @jiquem
> 
> Should every light switch be paired with a new empty group?

Button have to be paired on empty group, not lights...
This change the behaviour of button send zigbee message

**Reply by villmatt** [reply]
Thanks heaps @jiquem 

I know this shouldnt be here, but i don’t know how else to contact you. 
I have a home with many legrand switches. Does your setup work well for you? What Zigbee dongle do you use?
I am using an SLZB-06M and Z2M, I am finding that many of my devices are taking a long time to respond, and i also have many failed messages when i try and turn lights on and off. Most of the time I have to try multiple times. I think its a latency issue since they are solid brick walls in the house. Zigbee map shows devices < 150 Link Quality. Any insight will be awesome. 

Happy for you or anyone with some insight to contact me via email if that is a better forum
mvil5595@hotmail.com

**Reply by jiquem** [reply]
Hello,
You can create one empty group and bind all switch to it. On my side I created the g_22m_fake empty group and bind 39 switches to it.
Regarding your other questions, I am using a conbee 2. I have from time to time (2 times last year) some crash, and I have to restart z2m. I plan to migrate to the Home Assistant Connect ZBT-1 key during next weeks.
